diff --git a/build-libssl.sh b/build-libssl.sh
index bfb21f6..a9a3dcd 100755
--- a/build-libssl.sh
+++ b/build-libssl.sh
@@ -128,9 +128,9 @@ run_configure()
   echo "  Configure..."
   set +e
   if [ "${LOG_VERBOSE}" == "verbose" ]; then
-    ./Configure ${LOCAL_CONFIG_OPTIONS} no-tests | tee "${LOG}"
+    ./Configure ${LOCAL_CONFIG_OPTIONS} -DOPENSSL_API_COMPAT=0x00908000L -DOPENSSL_MIN_API==0x00908000L --api=0.9.8 no-tests | tee "${LOG}"
   else
-    (./Configure ${LOCAL_CONFIG_OPTIONS} no-tests > "${LOG}" 2>&1) & spinner
+    (./Configure ${LOCAL_CONFIG_OPTIONS} -DOPENSSL_API_COMPAT=0x00908000L -DOPENSSL_MIN_API==0x00908000L --api=0.9.8 no-tests > "${LOG}" 2>&1) & spinner
   fi
 
   # Check for error status
@@ -161,11 +161,15 @@ finish_build_loop()
   # Add references to library files to relevant arrays
   if [[ "${PLATFORM}" == AppleTV* ]]; then
     LIBSSL_TVOS+=("${TARGETDIR}/lib/libssl.a")
+    LIBSSL_TVOS_DYNAMIC+=("${TARGETDIR}/lib/libssl.dylib")
     LIBCRYPTO_TVOS+=("${TARGETDIR}/lib/libcrypto.a")
+    LIBCRYPTO_TVOS_DYNAMIC+=("${TARGETDIR}/lib/libcrypto.dylib")
     OPENSSLCONF_SUFFIX="tvos_${ARCH}"
   else
     LIBSSL_IOS+=("${TARGETDIR}/lib/libssl.a")
+    LIBSSL_IOS_DYNAMIC+=("${TARGETDIR}/lib/libssl.dylib")
     LIBCRYPTO_IOS+=("${TARGETDIR}/lib/libcrypto.a")
+    LIBCRYPTO_IOS_DYNAMIC+=("${TARGETDIR}/lib/libcrypto.dylib")
     if [[ "${PLATFORM}" != MacOSX* ]]; then
       OPENSSLCONF_SUFFIX="ios_${ARCH}"
     else
@@ -446,6 +450,11 @@ LIBCRYPTO_IOS=()
 LIBSSL_TVOS=()
 LIBCRYPTO_TVOS=()
 
+LIBSSL_IOS_DYNAMIC=()
+LIBCRYPTO_IOS_DYNAMIC=()
+LIBSSL_TVOS_DYNAMIC=()
+LIBCRYPTO_TVOS_DYNAMIC=()
+
 # Run relevant build loop
 source "${SCRIPTDIR}/scripts/build-loop-targets.sh"
 
@@ -454,6 +463,10 @@ if [ ${#LIBSSL_IOS[@]} -gt 0 ]; then
   echo "Build library for iOS..."
   lipo -create ${LIBSSL_IOS[@]} -output "${CURRENTPATH}/lib/libssl.a"
   lipo -create ${LIBCRYPTO_IOS[@]} -output "${CURRENTPATH}/lib/libcrypto.a"
+
+  lipo -create ${LIBSSL_IOS_DYNAMIC[@]} -output "${CURRENTPATH}/lib/libssl.dylib"
+  lipo -create ${LIBCRYPTO_IOS_DYNAMIC[@]} -output "${CURRENTPATH}/lib/libcrypto.dylib"
+
   echo "\n=====>iOS SSL and Crypto lib files:"
   echo "${CURRENTPATH}/lib/libssl.a"
   echo "${CURRENTPATH}/lib/libcrypto.a"
@@ -464,6 +477,10 @@ if [ ${#LIBSSL_TVOS[@]} -gt 0 ]; then
   echo "Build library for tvOS..."
   lipo -create ${LIBSSL_TVOS[@]} -output "${CURRENTPATH}/lib/libssl-tvOS.a"
   lipo -create ${LIBCRYPTO_TVOS[@]} -output "${CURRENTPATH}/lib/libcrypto-tvOS.a"
+
+  lipo -create ${LIBSSL_TVOS_DYNAMIC[@]} -output "${CURRENTPATH}/lib/libssl-tvOS.dylib"
+  lipo -create ${LIBCRYPTO_TVOS_DYNAMIC[@]} -output "${CURRENTPATH}/lib/libcrypto-tvOS.dylib"
+
   echo "\n=====>tvOS SSL and Crypto lib files:"
   echo "${CURRENTPATH}/lib/libssl-tvOS.a"
   echo "${CURRENTPATH}/lib/libcrypto-tvOS.a"
diff --git a/scripts/build-loop-targets.sh b/scripts/build-loop-targets.sh
index d019942..9a7442e 100755
--- a/scripts/build-loop-targets.sh
+++ b/scripts/build-loop-targets.sh
@@ -64,7 +64,7 @@ do
   # Add build target, --prefix and prevent async (references to getcontext(),
   # setcontext() and makecontext() result in App Store rejections) and creation
   # of shared libraries (default since 1.1.0)
-  LOCAL_CONFIG_OPTIONS="${TARGET} --prefix=${TARGETDIR} ${CONFIG_OPTIONS} no-async no-shared"
+  LOCAL_CONFIG_OPTIONS="${TARGET} --prefix=${TARGETDIR} ${CONFIG_OPTIONS} no-async shared"
 
   # Only relevant for 64 bit builds
   if [[ "${CONFIG_ENABLE_EC_NISTP_64_GCC_128}" == "true" && "${ARCH}" == *64  ]]; then
