--- OpenSSL-for-iPhone/include/openssl/crypto.h.ORIG	2020-02-19 09:54:32.000000000 -0500
+++ OpenSSL-for-iPhone/include/openssl/crypto.h	2020-02-19 10:09:31.000000000 -0500
@@ -46,6 +44,8 @@
 extern "C" {
 #endif
 
+#define OPENSSL_API_COMPAT 0x00908000L
+
 # if OPENSSL_API_COMPAT < 0x10100000L
 #  define SSLeay                  OpenSSL_version_num
 #  define SSLeay_version          OpenSSL_version
