--- OpenSSL-for-iPhone/include/openssl/ssl.h.ORIG	2020-02-19 09:55:08.000000000 -0500
+++ OpenSSL-for-iPhone/include/openssl/ssl.h	2020-02-19 10:04:52.000000000 -0500
@@ -12,6 +12,8 @@
 #ifndef HEADER_SSL_H
 # define HEADER_SSL_H
 
+#define OPENSSL_API_COMPAT 0x00908000L
+
 # include <openssl/e_os2.h>
 # include <openssl/opensslconf.h>
 # include <openssl/comp.h>
@@ -35,6 +37,8 @@
 extern "C" {
 #endif
 
+#define OPENSSL_API_COMPAT 0x00908000L
+
 /* OpenSSL version number for ASN.1 encoding of the session information */
 /*-
  * Version 0 - initial version
