diff --git a/CMakeLists.txt b/CMakeLists.txt
index c4334f5..dc07a67 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -529,9 +529,9 @@ endif()
 
 set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
 
-if(NOT IOS)
-	find_package(Threads REQUIRED)
-endif()
+#if(NOT IOS)
+#	find_package(Threads REQUIRED)
+#endif()
 
 if(NOT WIN32)
 	CHECK_SYMBOL_EXISTS(pthread_mutex_timedlock pthread.h HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL)
