diff --git a/scripts/android-build-openh264.sh b/scripts/android-build-openh264.sh
index 5e9abed2c..8e8cb7425 100755
--- a/scripts/android-build-openh264.sh
+++ b/scripts/android-build-openh264.sh
@@ -9,7 +9,7 @@ function build {
 	BASE=$(pwd)
 	common_run cd $BUILD_SRC
 	PATH=$ANDROID_NDK:$PATH
-	MAKE="make PATH=$PATH ENABLEPIC=Yes OS=android NDKROOT=$ANDROID_NDK TARGET=android-$2 NDKLEVEL=$2 ARCH=$1 -j libraries"
+	MAKE="make PATH=$PATH ENABLEPIC=Yes OS=android NDKROOT=$OPENH264_NDK TARGET=android-$2 NDKLEVEL=$2 ARCH=$1 -j libraries"
 	
 	common_run export QUIET_AR="$CCACHE "
 	common_run export QUIET_ASM="$CCACHE "
