diff --git a/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java b/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java
index bab6345d3..a2e59f9cd 100644
--- a/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java
+++ b/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java
@@ -277,7 +277,9 @@ public class LibFreeRDP
 		args.add(addFlag("themes", flags.getTheming()));
 		args.add(addFlag("fonts", flags.getFontSmoothing()));
 		args.add(addFlag("aero", flags.getDesktopComposition()));
-		args.add(addFlag("glyph-cache", false));
+		//args.add(addFlag("glyph-cache", false));
+		args.add(addFlag("glyph-cache", true));
+		args.add(addFlag("relax-order-checks", true));
 
 		if (!advanced.getRemoteProgram().isEmpty())
 		{
