diff --git a/client/Android/android_freerdp.c b/client/Android/android_freerdp.c
index bab8f80c7..81c8a1870 100644
--- a/client/Android/android_freerdp.c
+++ b/client/Android/android_freerdp.c
@@ -997,7 +997,7 @@ static jboolean JNICALL jni_freerdp_send_clipboard_data(JNIEnv* env, jclass cls,
 	ANDROID_EVENT* event;
 	freerdp* inst = (freerdp*)instance;
 	const jbyte* data = jdata != NULL ? (*env)->GetStringUTFChars(env, jdata, NULL) : NULL;
-	const size_t data_length = data ? (*env)->GetStringUTFLength(env, data) : 0;
+	int data_length = data ? strlen(data) : 0;
 	jboolean ret = JNI_FALSE;
 	event = (ANDROID_EVENT*)android_event_clipboard_new((void*)data, data_length);
 
