diff --git a/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java b/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java
index 0f4a92188..441234a7f 100644
--- a/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java
+++ b/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java
@@ -22,6 +22,7 @@ import com.freerdp.freerdpcore.domain.BookmarkBase;
 import com.freerdp.freerdpcore.domain.ManualBookmark;
 import com.freerdp.freerdpcore.presentation.ApplicationSettingsActivity;
 
+import java.io.File;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
@@ -332,8 +333,11 @@ public class LibFreeRDP
 
 		if (advanced.getRedirectSDCard())
 		{
-			String path = android.os.Environment.getExternalStorageDirectory().getPath();
-			args.add("/drive:sdcard," + path);
+			File dir = context.getExternalFilesDir(null);
+			if (dir != null) {
+				String path = dir.getPath();
+				args.add("/drive:sdcard," + path);
+			}
 		}
 
 		args.add("/clipboard");
