diff --git a/scripts/android-build-freerdp.sh b/scripts/android-build-freerdp.sh
index b27fa6c9b..1efbcdad5 100755
--- a/scripts/android-build-freerdp.sh
+++ b/scripts/android-build-freerdp.sh
@@ -105,6 +105,15 @@ do
     else
         CMAKE_CMD_ARGS="$CMAKE_CMD_ARGS -DWITH_JPEG=OFF"
     fi
+
+    CMAKE_CMD_ARGS="$CMAKE_CMD_ARGS -DWITH_NEON=STATE"
+    if [ "$ARCH" == "armeabi-v7a" ];
+    then
+        CMAKE_CMD_ARGS="$(echo $CMAKE_CMD_ARGS | sed -i 's/WITH_NEON=STATE/WITH_NEON=OFF/')"
+    else
+        CMAKE_CMD_ARGS="$(echo $CMAKE_CMD_ARGS | sed -i 's/WITH_NEON=STATE/WITH_NEON=ON/')"
+    fi
+
     if [ $WITH_OPENH264 -ne 0 ];
     then
         if [ $BUILD_DEPS -ne 0 ];
