diff --git a/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java b/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java
index 639383c7f..4f421d6e0 100644
--- a/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java
+++ b/client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java
@@ -26,12 +26,19 @@ import com.freerdp.freerdpcore.presentation.ApplicationSettingsActivity;
 import java.io.File;
 import java.util.ArrayList;
 import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class LibFreeRDP
 {
 	private static final String TAG = "LibFreeRDP";
+	public static final int HAS_H264_SETTER_TIMEOUT = 1;
 	private static EventListener listener;
 	private static boolean mHasH264 = false;
 
@@ -65,6 +72,7 @@ public class LibFreeRDP
 				{
 		return tryLoad(new String[] { library });
 				}
+
 	static
 	{
 		try
@@ -81,11 +89,11 @@ public class LibFreeRDP
 			int patch = Integer.parseInt(Objects.requireNonNull(matcher.group(3)));
 
 			if (major > 2)
-				mHasH264 = freerdp_has_h264();
+				setmHasH264();
 			else if (minor > 5)
-				mHasH264 = freerdp_has_h264();
+				setmHasH264();
 			else if ((minor == 5) && (patch >= 1))
-				mHasH264 = freerdp_has_h264();
+				setmHasH264();
 			else
 				throw new RuntimeException("APK broken: native library version " + version +
 				                           " does not meet requirements!");
@@ -99,6 +107,20 @@ public class LibFreeRDP
 		}
 	}
 
+	private static void setmHasH264() {
+		final ExecutorService mHasH264Executor = Executors.newSingleThreadExecutor();
+		final Future<?> future = mHasH264Executor.submit(() -> mHasH264 = freerdp_has_h264());
+		mHasH264Executor.shutdown();
+		try {
+			future.get(HAS_H264_SETTER_TIMEOUT, TimeUnit.SECONDS);
+		}
+		catch (ExecutionException|InterruptedException|TimeoutException e) {
+			Log.e(TAG, Log.getStackTraceString(e));
+		}
+		if (!mHasH264Executor.isTerminated())
+			mHasH264Executor.shutdownNow();
+	}
+
 	public static boolean hasH264Support()
 	{
 		return mHasH264;
