--- spice-gtk-0.37/src/channel-main.c.orig	2020-08-24 23:48:27.670981760 -0400
+++ spice-gtk-0.37/src/channel-main.c	2020-08-24 23:51:01.994949872 -0400
@@ -2019,8 +2019,8 @@
             c->agent_display_config_sent = true;
         }
 
-        agent_sync_audio_playback(self);
-        agent_sync_audio_record(self);
+        //agent_sync_audio_playback(self);
+        //agent_sync_audio_record(self);
 
         agent_max_clipboard(self);
 
