diff -urN spice-gtk-0.32/spice-common/tests/test-logging.c spice-gtk/spice-common/tests/test-logging.c
--- spice-gtk-0.32/spice-common/tests/test-logging.c	2016-05-30 12:17:44.000000000 -0400
+++ spice-gtk/spice-common/tests/test-logging.c	2016-08-25 13:44:20.434768477 -0400
@@ -366,7 +366,7 @@
 
 static void handle_sigabrt(int sig)
 {
-    _Exit(1);
+    _exit(1);
 }
 
 int main(int argc, char **argv)
